### ISSUE TITLE
Unexpected static keyword

### ISSUE THREAD

**Issue by chaewan-park** [nicklockwood/SwiftFormat#1592]
With this code:

```swift
public final class MyClass {
    private static func privateStaticFunction1() -> Bool {
        switch Result(catching: { try someThrowingFunction() }) {
        case .success:
            return true
        case .failure:
            return false
        }
    }
    
    private static func privateStaticFunction2() -> Bool { // <-- Unexpected static keyword
        return false
    }
}
```
Running `swiftformat` MyClass.swift fails with:

```
error: Unexpected static keyword on line 11 in MyClass.swift
```

And my `.swiftformat` file has this option:
```
# format options
--self insert
```
using 0.52.10

**Comment by nicklockwood** [comment]
@chaewan-park I'm not able to reproduce this in SwiftFormat 0.52.10 with the following unit test. Perhapse there's some other surrounding context in the file needed to recreate it? Or is it possible you somehow have multiple SwiftFormat versions installed, and are running the wrong one?

```swift
    func testRedundantSelfNotConfusedByStaticAfterSwitch() {
        let input = """
        public final class MyClass {
            private static func privateStaticFunction1() -> Bool {
                switch Result(catching: { try someThrowingFunction() }) {
                case .success:
                    return true
                case .failure:
                    return false
                }
            }

            private static func privateStaticFunction2() -> Bool { // <-- Unexpected static keyword
                return false
            }
        }
        """
        testFormatting(for: input, rule: FormatRules.indent, exclude: ["enumNamespaces"])
    }
```

**Comment by nicklockwood** [comment]
@chaewan-park fixed in 0.52.11
